### PR TITLE
Git Hub Action - downstream-test-go

### PR DESCRIPTION
**Pull request by n3wscott**
adding a repo to create a custom github action for knative to test downstream repos upgradability from upstream repos.

### CODE REVIEW

**Comment by grantr** [comment]
(Still working out the self-serve process here, @mattmoor #241)

What WG will own this repo? Should they be added as writers?

**Comment by n3wscott** [comment]
> What WG will own this repo? Should they be added as writers?

Good question, I suppose it should be productivity.

**Comment by chizhg** [comment]
Could you please provide a bit more context for this repo? Is it supposed to be working together with https://github.com/knative-sandbox/integration?

**Comment by n3wscott** [comment]
this repo will turn https://github.com/knative/pkg/pull/1653 into a github action to allow us to use it with less copy/paste

**Comment by n3wscott** [comment]
I have an action staged that should replace knative/pkg#1653 with:

```yaml
- uses: knative-sandbox/downstream-test-go@v1
  with:
    upstream-module: 'knative.dev/pkg'
    downstream-repository: 'knative-sandbox/sample-controller'
    downstream-module: 'knative.dev/sample-controller'
```

**Comment by chizhg** [comment]
@n3wscott If I understand correctly, is the intention of this GitHub action to prevent the upstream (`knative/pkg`) change breaking the tests in downstream (`knative-sandbox/sample-controller`) ?

**Comment by n3wscott** [comment]
> @n3wscott Not sure if I understand it correctly, is the intention of this GitHub action to prevent the upstream (`knative/pkg`) change breaking the tests in downstream (`knative-sandbox/sample-controller`) ?

We will not block PRs in the upstream, but we are trying to give info to the reviewers or PR authors that this PR will break downstream repos.

**Comment by chizhg** [comment]
> > @n3wscott Not sure if I understand it correctly, is the intention of this GitHub action to prevent the upstream (`knative/pkg`) change breaking the tests in downstream (`knative-sandbox/sample-controller`) ?
> 
> We will not block PRs in the upstream, but we are trying to give info to the reviewers or PR authors that this PR will break downstream repos.

Thanks for the explanation! It looks like something @Cynocracy is very interested in.

File: peribolos/knative-sandbox.yaml [hunk 1]
```diff
@@ -443,6 +450,7 @@ orgs:
         privacy: closed
         repos:
           .github: write
+          downstream-test-go: admin
```
Comment (mattmoor) [inline peribolos/knative-sandbox.yaml:453]: Why more than `write`?

Comment (n3wscott) [reply peribolos/knative-sandbox.yaml:453]: oh, sorry... copy/pasted will fix

**Pull request merged**